core/arch/arches_test.go

```diff
@@ -25,22 +24,45 @@ func (s archSuite) TestContains(c *gc.C) {
 
 func (s archSuite) TestStringList(c *gc.C) {
 	arches := arch.AllArches()
-	c.Assert(arches.StringList(), jc.DeepEquals, []string{"amd64", "arm64", "armhf", "i386", "ppc64el", "riscv64", "s390x"})
+	c.Assert(arches.StringList(), jc.DeepEquals, []string{"amd64", "arm64", "ppc64el", "riscv64", "s390x"})
 }
 
 func (s archSuite) TestString(c *gc.C) {
 	arches := arch.AllArches()
-	c.Assert(arches.String(), gc.Equals, "amd64,arm64,armhf,i386,ppc64el,riscv64,s390x")
+	c.Assert(arches.String(), gc.Equals, "amd64,arm64,ppc64el,riscv64,s390x")
 }
 
-func (s archSuite) TestConstraintArch(c *gc.C) {
-	a := arch.ConstraintArch(constraints.MustParse("mem=4G"), nil)
-	c.Assert(a, gc.Equals, "amd64")
-	a = arch.ConstraintArch(constraints.MustParse("arch=arm64"), nil)
-	c.Assert(a, gc.Equals, "arm64")
-	defaultCons := constraints.MustParse("arch=arm64")
-	a = arch.ConstraintArch(constraints.MustParse("mem=4G"), &defaultCons)
-	c.Assert(a, gc.Equals, "arm64")
-	a = arch.ConstraintArch(constraints.MustParse("arch=s390x"), &defaultCons)
-	c.Assert(a, gc.Equals, "s390x")
+func (s *archSuite) TestHostArch(c *gc.C) {
+	a := arch.HostArch()
+	c.Assert(arch.IsSupportedArch(a), jc.IsTrue)
+}
+
+func (s *archSuite) TestNormaliseArch(c *gc.C) {
+	for _, test := range []struct {
+		raw  string
+		arch string
+	}{
+		{"windows", "windows"},
+		{"amd64", "amd64"},
+		{"x86_64", "amd64"},
+		{"aarch64", "arm64"},
+		{"arm64", "arm64"},
+		{"ppc64el", "ppc64el"},
+		{"ppc64le", "ppc64el"},
+		{"s390x", "s390x"},
+		{"riscv64", "riscv64"},
+		{"risc", "riscv64"},
```

Comment:
I'm unsure about this one, as we do move it to `riscv64`, I'm not a fan of dropping the word size, I think we should be a lot stricter.